device_doctor/lib/src/process_helper.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show utf8;
+import 'dart:io';
+
+import 'utils.dart';
+
+List<ProcessInfo> _runningProcesses = <ProcessInfo>[];
+
+/// Defines detailed information of a process.
+class ProcessInfo {
+  ProcessInfo(this.command, this.process);
+
+  final DateTime startTime = DateTime.now();
+  final String command;
+  final Process process;
+
+  @override
+  String toString() {
+    return '''
+  command : $command
+  started : $startTime
+  pid     : ${process.pid}
+'''
+        .trim();
+  }
+}
+
+/// Starts a process for an executable command, and returns the processes.
+Future<Process> startProcess(String executable, List<String> arguments,
+    {Map<String, String> env, bool silent: false}) async {
+  String command = '$executable ${arguments?.join(" ") ?? ""}';
+  if (!silent) logger.info('Executing: $command');
+  Process proc =
+      await processManager.start(<String>[executable]..addAll(arguments), environment: env, workingDirectory: cwd);
+  ProcessInfo procInfo = ProcessInfo(command, proc);
+  _runningProcesses.add(procInfo);
+
+  // ignore: unawaited_futures
+  proc.exitCode.then((_) {
+    _runningProcesses.remove(procInfo);
+  });
+
+  return proc;
+}
+
+/// Executes a command and returns its standard output as a String.
+///
+/// Standard error is redirected to the current process' standard error stream.
+Future<String> eval(String executable, List<String> arguments,
```

Comment:
I hesitate from `eval` as it's more associated with dynamic code calling, but here we know what we're calling. `execCommand`?

Reply:
As this is just the initial step of the tool, we will have more general cases on the way, and we also will implement exec as well.

Reply:
https://unix.stackexchange.com/questions/296838/whats-the-difference-between-eval-and-exec

What would be the difference between the exec and eval in this project?

Reply:
See https://github.com/flutter/cocoon/blob/master/agent/lib/src/utils.dart#L304 and https://github.com/flutter/cocoon/blob/master/agent/lib/src/utils.dart#L336

Reply:
Good point. Returning ProcessResult should deduplicate that logic

Reply:
Will consider consolidating when `exec` is called in ongoing PRs.